src/sentry/relay/config/metric_extraction.py

```diff
@@ -533,26 +519,29 @@ def _convert_aggregate_and_query_to_metrics(
         if not should_use_on_demand_metrics(dataset, aggregate, query, groupbys, prefilling):
             return None
 
-        on_demand_spec = OnDemandMetricSpec(
-            field=aggregate,
-            query=query,
-            environment=environment,
-            groupbys=groupbys,
-            spec_type=spec_type,
-            use_updated_env_logic=use_updated_env_logic,
-        )
-
-        metric_spec = on_demand_spec.to_metric_spec(project)
-        # TODO: switch to validate_rule_condition
-        if (condition := metric_spec.get("condition")) is not None:
-            validate_sampling_condition(json.dumps(condition))
-        else:
-            metrics.incr(
-                "on_demand_metrics.missing_condition_spec",
-                tags={"prefilling": prefilling},
+        metric_specs_and_hashes = []
+        # Create as many specs as we support
+        for spec_version in OnDemandMetricSpecVersioning.get_spec_versions():
+            on_demand_spec = OnDemandMetricSpec(
```

Comment:
This block is right-shifted and we now pass `spec_version=spec_version` to this object.